frontend/src/AppContext.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { createContext, Dispatch, useReducer } from 'react';
+import {
+  getLinkUrls,
+  NormalizedPractitioner,
+  normalizeFetchedPractitioners,
+  normalizePractitionerResult
+} from './utils/helpers';
+
+export interface IAppContextData {
+  practitioners: NormalizedPractitioner[];
+  previousPage: string | null,
+  nextPage: string | null,
+  fetched: boolean;
+}
+
+interface IAppContext {
+  state: IAppContextData;
+  dispatch: Dispatch<any>
+}
+
+let initialState: IAppContextData = {
+  practitioners: [],
+  previousPage: null,
+  nextPage: null,
+  fetched: false,
+}
+
+const AppContext = createContext<IAppContext | null>(null);
+
+let reducer = (state, action) => {
```

Comment:
Love that you are using reducers here. Seems like the best choice, given the complexity we are dealing with. And my preference is to avoid state management libraries until you can clearly justify them. This app feels like it deserves a lighter approach.

Reply:
totally agree. i ran into a point where i was going to have to do a put request and normalize of data and thought it would be easier to have context do that instead of storing the results in a hook.